### ISSUE TITLE
IsString instances

### ISSUE THREAD

**Issue by centromere** [vincenthz/hs-memory#21]
The more I use the library, the more I wish `ScrubbedBytes` derived an instance of IsString. Can this be accommodated? A use case is, for example, when I am unit testing and need to embed test vectors directly in the code.

**Comment by vincenthz** [comment]
yes, I think it's possible. there's couple of caveat though:
- ScrubbedBytes cannot point to the executable's memory directly, otherwise it will try to wipe it on GC.
- The executable memory is not in a binary format anyway (serialized in modified UTF-8 encoding)

I can add an in-efficient one (that use `[Char]`) to get started, and then we can add a rules to go from modified UTF-8 to binary much faster (`unpackCStringUtf8#`)

edit: also I've looked into doing this before, but when I looked at how `bytestring` did it with a RULES, made me worry of NUL bytes and correct serialization, as the `bytestring` rules use `c_strlen`. it turns out this rules is not triggered anymore.
